src/Compiler/Checking/PostInferenceChecks.fs

```diff
@@ -1143,7 +1145,7 @@ and CheckExpr (cenv: cenv) (env: env) origExpr (ctxt: PermitByRefExpr) : Limit =
     | Expr.Let _ 
     | Expr.Sequential (_, _, NormalSeq, _)
     | Expr.DebugPoint _ -> 
-        CheckExprLinear cenv env expr ctxt id
+        CheckExprLinear cenv env expr ctxt NoLimit
```

Comment:
This is the line that could be wrong, given that `NoLimit` is not a true zero for `CombineTwoLimits`.

Reply:
Sorry, I used the wrong words. It's not that it necessarily can go wrong; what I meant to say is that there are possible implementations of CombineTwoLimits for which this refactor could change semantics, and I haven't yet thought hard enough to convince myself that the semantics are the same in this case.

Reply:
Concretely, in the previous world the base value was decided by something inside the loop, whereas now it's decided by the caller at the top level. They're probably the same base value, but I haven't yet convinced myself of this.